File: leetcode/src/test/java/problems/no6/SolutionTest.java
package problems.no6;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

class SolutionTest {

    private final Solution solution = new Solution();

    @Test
    void 기본_테스트_케이스_1() {

        String s = "PAYPALISHIRING";
        int numRows = 3;

        assertThat("PAHNAPLSIIGYIR", is(equalTo(solution.convert(s, numRows))));
    }


    @Test
    void 기본_테스트_케이스_2() {

        String s = "PAYPALISHIRING";
        int numRows = 4;

        assertThat("PINALSIGYAHRPI", is(equalTo(solution.convert(s, numRows))));
    }

    @Test
    void 기본_테스트_케이스_3() {

        String s = "A";
        int numRows = 1;

        assertThat("A", is(equalTo(solution.convert(s, numRows))));
    }


    @Test
    void 기본_테스트_케이스_4() {

        String s = "AB";
        int numRows = 1;

        assertThat("AB", is(equalTo(solution.convert(s, numRows))));
    }

}

File: leetcode/src/test/kotlin/problems/no9/SolutionTest.kt
package problems.no9

import org.junit.jupiter.api.Assertions.*

import org.hamcrest.CoreMatchers.`is`
import org.hamcrest.CoreMatchers.equalTo
import org.hamcrest.MatcherAssert.assertThat
import org.junit.jupiter.api.Test

internal class SolutionTest {
    private val solution: Solution = Solution()

    @Test
    fun 테스트_케이스_1() {
        val input = 121
        val output = true
        assertThat(output, `is`(equalTo(solution.isPalindrome(input))))
    }

    @Test
    fun 테스트_케이스_2() {
        val input = -121
        val output = false
        assertThat(output, `is`(equalTo(solution.isPalindrome(input))))
    }

    @Test
    fun 테스트_케이스_3() {
        val input = 10
        val output = false
        assertThat(output, `is`(equalTo(solution.isPalindrome(input))))
    }

    @Test
    fun 테스트_케이스_4() {
        val input = -101
        val output = false
        assertThat(output, `is`(equalTo(solution.isPalindrome(input))))
    }

}